tests/lib/test_config.py

```diff
@@ -1,176 +1,206 @@
 import nestor_api.lib.config as config
-import nestor_api.lib.io as io
-
-
-def test_change_environment(mocker):
-    spy = mocker.patch.object(io, "execute")
-
-    config.change_environment("environment", "path/to/config")
-
-    spy.assert_has_calls(
-        [
-            call("git stash", "path/to/config"),
-            call("git fetch origin", "path/to/config"),
-            call("git checkout environment", "path/to/config"),
-            call("git reset --hard origin/environment", "path/to/config"),
-        ]
-    )
-
-
-@pytest.mark.usefixtures("config")
-def test_create_temporary_config_copy(mocker):
-    spy = mocker.patch.object(io, "create_temporary_copy", return_value="/temporary/path")
-
-    path = config.create_temporary_config_copy()
-
-    spy.assert_called_once_with("/fixtures-nestor-config", "config")
-    assert path == "/temporary/path"
-
-
-def test_get_app_config(mocker):
-    mocker.patch.object(Configuration, "get_config_path", return_value="tests/__fixtures__/config")
 
-    app_config = config.get_app_config("backoffice")
+FAKE_ENV = {
+    "NESTOR_CONFIG_PATH": "/fixtures-nestor-config",
+    "NESTOR_PRISTINE_PATH": "/fixtures-nestor-pristine",
+    "NESTOR_WORK_PATH": "/fixtures-nestor-work",
+}
+
+
+@patch("nestor_api.lib.config.io", autospec=True)
+# pylint: disable=no-self-use
+class TestConfigLibrary(unittest.TestCase):
+    def test_change_environment(self, io_mock):
+        config.change_environment("environment", "path/to/config")
+
+        io_mock.execute.assert_has_calls(
+            [
+                call("git stash", "path/to/config"),
+                call("git fetch origin", "path/to/config"),
+                call("git checkout environment", "path/to/config"),
+                call("git reset --hard origin/environment", "path/to/config"),
+            ]
+        )
 
-    assert app_config == {
-        "domain": "website.com",
-        "sub_domain": "backoffice",
-        "variables": {
-            "ope": {
-                "VARIABLE_OPE_1": "ope_1",
-                "VARIABLE_OPE_2": "ope_2_override",
-                "VARIABLE_OPE_3": "ope_3",
+    @patch.dict("os.environ", FAKE_ENV)
+    def test_create_temporary_config_copy(self, io_mock):
+        io_mock.create_temporary_copy.return_value = "/temporary/path"
+
+        path = config.create_temporary_config_copy()
+
+        io_mock.create_temporary_copy.assert_called_once_with("/fixtures-nestor-config", "config")
+        assert path == "/temporary/path"
+
+    @patch("nestor_api.lib.config.get_project_config", autospec=True)
+    @patch("nestor_api.lib.config.Configuration", autospec=True)
+    def test_get_app_config(self, configuration_mock, get_project_config_mock, io_mock):
+        configuration_mock.get_config_path.return_value = "tests/__fixtures__/config"
+        configuration_mock.get_config_app_folder.return_value = "apps"
+        io_mock.exists.return_value = True
+        io_mock.from_yaml.return_value = {
```

Comment:
I think that this is something we should talk about and all agree on, but I find that these "mock configurations" are a bit polluting the test code. 
There is an [interesting discussion about the options to mock multiple methods ](https://stackoverflow.com/questions/29318987/preferred-way-of-patching-multiple-methods-in-python-unit-test).

Personally I think that I would be more in favor of using 
```python
@patch.multiple('nestor_api.lib.config.Configuration', 
                get_config_path=MagicMock(return_value='tests/__fixtures__/config'), 
                get_config_app_folder=MagicMock(return_value='apps'))
```

Let me know what you think

Reply:
I understand your point but I think it is just moving the problem to decorators that will also clutter the code and if you have a return value that is quite bit (a dict for example), then it would be a little bit weird to have it in the patch section :/ 
I know that fixture in pytest are definitely better approach for this point... 🙄 

Usually when the test function body become important I used to separate them with comment to make it clearer like this:
```python
def myFunction():
	# Mocks
    ...
	# Test
	...
	# Assertions
	...
```
What do you think ?

Reply:
I think that it would not clutter the code that much if big return values are fixtures (either in another file or just a variable in the test file).
But this brings another debate / problem of not having fixtures right next to the test case.

Yes, at least comments, imo, so that it is easy to skim through the test

Reply:
Ok I've gone with comments but understand your point and we should discuss about it. I create a note in the board :)